imagery/i.gensigset/subcluster.c

```diff
@@ -366,10 +366,14 @@ static double regroup(struct ClassSig *Sig, int nbands)
             subsum += tmp;
             Data->p[s][i] = tmp;
         }
-        likelihood += log(subsum) + maxlike;
 
-        for (i = 0; i < Sig->nsubclasses; i++)
-            Data->p[s][i] /= subsum;
+        if (subsum > 0)
+            likelihood += log(subsum) + maxlike;
+
+        if (subsum) {
```

Comment:
Didn't combine both of these in the same conditional as `subsum` being negative is not valid for `log`, but valid while calculating `Data`.